mitodl/ocw-studio
# redirect to login on authentication failures

#### Pre-Flight checklist


#### What are the relevant tickets?
Together with #1298, this will complete #1177

#### What's this PR do?
This PR prompts user to log in when any API rejects with an authentication error.

#### How should this be manually tested?
1. **Optionally:** *You can test that you get redirected to login without actually setting this up*. Set up saml login as described in the readme (https://github.com/mitodl/ocw-studio#testing-touchstone-login-with-saml-via-ssocircle)
2. Login in to studio (as superuser or via saml)
3. Navigate to any site page
4. Delete your `sessionid` cookie.

    _In chrome, you can do this in dev console --> Application_

    <img width="400" alt="Screen Shot 2022-05-05 at 11 14 12 AM" src="https://user-images.githubusercontent.com/9010790/166955493-52890a89-ae5c-4c54-b325-f3afae889d0f.png">

5. Click on any other page that triggers an API call. So...almost anything.
6. You should be presented with a dialog indicating your session has expired.
    - you can ignore it by clicking "Cancel"
    - or you can go to login by clicking... "Go to Login".
7. If you ignore the dialog, you should be re-prompted when future API requests fail with authentication errors.
8. Errors not related to authentication should not trigger the dialog. **I'm not sure the easiest way to trigger these. If you really want to, you can log in as a user that does not have site creation permission and try creating a site.**

#### Where should the reviewer start?
⚠️ When reviewing the code, I would encourage you to review commit-by-commit. A fairly large fraction of the line changes are in the first 5 commits, which are mostly renaming / removing confusing dead code.
